### ISSUE TITLE
[Feature Request] DB Upgrade: new fields request

### ISSUE THREAD

**Issue by fernandog** [SiCKRAGETV/sickrage-issues#2589]
**TV_SHOWS**
- **size: store show size and only update with refresh and/or PP**
- **description**
- **actors**
- **search_method**

**TV_EPISODES**
- **torrent hash: auto remove torrent from client**
- **torrent client: where hash was sent**
- **torrent file link**
- **nzb guid**
- **nzb link**
- **actors**
- **watched**

- previous quality: if better quality stalls or fails, don't delete previous file and use previous quality
- previous status: same as above
- previous release file: if better quality stalls or fails, don't delete previous file and use previous quality. 
Also use this if user rename file and don't put all needed info

**history**
- **size**
- **hash**
- **link**

**Failed.db**
- **history.hash**
- **failed.hash**
<br/>
Include failed.db in sickbeard.db
Add quality to failed.db

Subtitles: add to DB all subtitles downloaded and all languages of them. Will fix flag not showing in history when enabling option to not append language to subtitle file and multi language enabled

Closed PR that needed DB upgrade:
https://github.com/SiCKRAGETV/SickRage/pulls?q=is%3Apr+label%3A%22DB+Version+Change%22+is%3Aclosed

**Comment by neoatomic** [comment]
Maybe the indexers that we (might want to) add in the future.?

Sportdb https://github.com/SiCKRAGETV/sickrage-issues/issues/872
Tv.com
And the split-off from TVRage witch name i forgot.

**Comment by fernandog** [comment]
I think don't need db upgrade. Just set sportdb as indexer 3 for example, tv.com as 4, etc..

**Comment by neoatomic** [comment]
And we have this old FR. Not sure is we would need an update for that. https://github.com/SiCKRAGETV/sickrage-issues/issues/750

**Comment by miigotu** [comment]
@fernandog we need a field "description" in tv_shows, and "actors" in both tv_shows and tv_episodes

Some of those wants above are not needed anymore, but we are just making a list right now. We can make final decisions later.

**Comment by WebSpider** [comment]
IMO it would be better to keep the torrent_hash in history, so we can make distinction between different download attempts (failed/proper/hung download) as well

**Comment by miigotu** [comment]
I figure if we have a few extra fields and end up not needing them, it is better than not having them and later needing them resulting in another db upgrade anytime soon.

**Comment by WebSpider** [comment]
Also, we are adding hash for torrent, is there anything similar to identify non-torrent downloads, maybe a unique id from provider or client?

**Comment by miigotu** [comment]
Nzb has a guid hash, similar to torrent hash, that will use the same hash field.

**Comment by WebSpider** [comment]
@neoatomic i dont think we'd need DB update for #750 .. we could simply download the posters as required and place them in images / cache/images, so we don't need to keep state.

We may want to look into what a full refresh does then.

For sports, we need to see. I've scanned the scene-rules for sports, and each sport has clearly documented, but massively different scene-rules. We could just compare all the sports regexes, or limit it to the type of sports we're talking about. In the latter case, we would have to store the sports-type.

Also: No cache.db updates needed, when storing the hash?

**Comment by miigotu** [comment]
I think hash/guid needs added in cache too, to help avoid extra processing later trying to determine it after the search.

**Comment by MGaetan89** [comment]
There is also https://github.com/SiCKRAGETV/sickrage-issues/issues/1642 that suggests a DB update to improve performance.

**Comment by WebSpider** [comment]
I'm unsure if indexes in sqlite are as useful as they are when using a SQL server. When using sqlite (not sqlite3), this resulted in full table scans anyway, and was only provided for compatibility with other SQL engines. Don't know how this progressed over the years though.

**Comment by miigotu** [comment]
@MGaetan89 @WebSpider those indices need gone over again and validated, as the SELECT and WHERE have changed in quite a few places in SR since then.

**Comment by WebSpider** [comment]
Also, would we store custom qualities in db? Or in config file? Db would make sense, because we store the downloaded quality in there as well

**Comment by miigotu** [comment]
@WebSpider the show table stores the quality it wants for the show, the episode table stores the quality it was downloaded.

For custom quality definitions that are defined by the user I think this is a user configuration, just like anything in the ini.

**Comment by WebSpider** [comment]
Also #1994

**Comment by fernandog** [comment]
Tv shows table:
Search_torrent
Search_nzb

Users can select if that show should be downloaded using torrent and/or nzb

**Comment by WebSpider** [comment]
Would prefer to make this field search_Method, which could be a dict. This would enable us to introduce new ways of searching without having to update db again.

-----Original Message-----

**Comment by miigotu** [comment]
yes a dict or bit field would work there. Also thinking about migrating the "enabled" field for providers to exactly that, so we can choose what is allowed to be searched on each provider. Allows selecting one provider for tv, one for anime, one for sports, etc. Or one provider can have a combo of each of those.

**Comment by fernandog** [comment]
tv episodes table: PAUSED / ARCHIVED column

related to this:
https://github.com/SiCKRAGETV/sickrage-issues/issues/510

**Comment by fernandog** [comment]
shouldn't we add this tv episodes suggestion to history or another new table?
torrent hash: auto remove torrent from client
torrent client: where hash was sent
torrent file link
nzb guid
nzb link

**Comment by fernandog** [comment]
no one?

**Comment by miigotu** [comment]
we don't need a paused or archived column once compound archived status is done.

**Comment by fernandog** [comment]
What about https://github.com/SiCKRAGETV/sickrage-issues/issues/2589#issuecomment-139415647 ?

One episode can have multiple torrents/guid

**Comment by miigotu** [comment]
for snatching a better status, the last hash for the last download would be in the history table. or in the failed table.

**Comment by fernandog** [comment]
so if user download 720p, and 10min later 1080p comes, SR would remove the 720p torrent/hash or guid (if users wants) and them add 1080p ?

**Comment by WebSpider** [comment]
I prefer to have a possibility to keep the snatch history, since we will need to know of each and every hash when we start improving client interaction

Technically we can do this in various ways: A new table referring to the episode, or just make the new hash column a dict stored as varchar/longtext or something... 

-----Original Message-----

**Comment by fernandog** [comment]
New setting in shows table:

'search only by name'

https://github.com/SiCKRAGETV/sickrage-issues/issues/965

**Comment by OmgImAlexis** [comment]
Skip multi-episode downloads.

https://github.com/SiCKRAGETV/sickrage-issues/issues/2953

**Comment by OmgImAlexis** [comment]
We have switch to using feathub for feature requests. If this feature is still important to anyone, please
check that it has not been added and add it [HERE](http://feathub.com/SiCKRAGETV/SickRage)
Make sure to vote!

**Comment by fernandog** [comment]
Add "Quality" column to failed.db

as discussed in old threads (can't find them)
avoid this N/A

![image](https://cloud.githubusercontent.com/assets/2620870/10119073/e44da1ac-6461-11e5-90f2-c64d288076a5.png)

**Comment by miigotu** [comment]
Some of these wont be necessary, we will have to go through these soon and decide once we get our current big tasks done.
